### PR TITLE
Allow null valued attribute definitions

### PR DESCRIPTION
**Pull request by kaandok**
implements #1193

### CODE REVIEW

File: core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java [hunk 1]
```diff
@@ -58,7 +58,9 @@ private SqlStatements(SqlStatements that) {
      * @return this
      */
     public SqlStatements define(String key, Object value) {
-        attributes.put(key, value);
+        if (value != null) {
+            attributes.put(key, value);
+        }
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java]: Should probably `remove(key)` if `value == null`.

Comment (kaandok) [reply core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java]: good idea, thank you.

**Comment by qualidafial** [comment]
The last commit is triggering a PMD rule violation:

```[INFO] PMD Failure: org.jdbi.v3.core.statement.SqlStatements:61 Rule:ConfusingTernary Priority:3 Avoid if (x != y) ..; else ```

Need to invert the if/else clause so it's:

```java
if (value == null) {
  attributes.remove(key);
} else {
  attributes.put(key, value);
}
```

**Comment by kaandok** [comment]
@qualidafial should be good to go now.

**Pull request merged**